src/ReturnTypes/ModelOnlyDynamicMethodReturnTypeExtension.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Larastan\Larastan\ReturnTypes;
+
+use Illuminate\Database\Eloquent\Model;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\Constant\ConstantArrayTypeBuilder;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\Type;
+
+use function array_map;
+use function array_filter;
+use function count;
+
+final class ModelOnlyDynamicMethodReturnTypeExtension implements DynamicMethodReturnTypeExtension
+{
+    public function getClass(): string
+    {
+        return Model::class;
+    }
+
+    public function isMethodSupported(MethodReflection $methodReflection): bool
+    {
+        return $methodReflection->getName() === 'only';
+    }
+
+    public function getTypeFromMethodCall(
+        MethodReflection $methodReflection,
+        MethodCall $methodCall,
+        Scope $scope
+    ): Type {
+        $fields = $methodCall->getArgs();
+
+        if (count($fields) < 1) {
+            return new ErrorType();
+        }
+
+        if ($fields[0]->value instanceof Array_) {
+            $fields = $fields[0]->value->items;
+        }
+
+        $fields = array_map(static fn ($f) => $f->value, array_filter($fields));
+
+        $model = $scope->getType($methodCall->var);
+        $array = ConstantArrayTypeBuilder::createEmpty();
+
+        foreach ($fields as $field) {
+            if (! $field instanceof String_) {
+                return new ArrayType(new MixedType(), new MixedType());
```

Comment:
same with this, would i not be better to just `continue` and ignore the field if it's not a string?

Reply:
so i just tested what laravel will do if given a non-string type to `only` and this is the behaviour:

```php
        dd($model->only(['test', 1]));
```

```
array:2 [
  "test" => null
  1 => null
]
```

Reply:
It's not just if it's not a string, what about variables or expressions?

E.g., 

```php
$model->only([
    ...$columns,
    $var,
]);
```

Reply:
I updated it to return `array<string, mixed>` if we can't determine all of the columns---I don't like skipping over fields we can't parse and returning an array shape that is incorrect.

Reply:
hmm that's an interesting point yeah;

you could use `$scope->getType($field)->getConstantStrings()` for the variadic or `$scope->getType($fields[0])->getItemType()->getConstantStrings()` for an array to determine if there are any known constant strings in the expression that is passed to only to try to be a bit more versatile, rather than relying on the AST node directly

that would allow you to properly detect unions of strings or other weird edge cases

Reply:
Done